scipy/stats/_distribution_infrastructure.py

```diff
@@ -911,6 +945,13 @@ def filtered(self, x, *args, **kwargs):
             any_endpoint = (mask_endpoint if mask_endpoint.shape == ()
                             else np.any(mask_endpoint))
 
+        # Check for non-integral arguments to PMF method
```

Comment:
Part of change 5: "Adjust the `_set_invalid_nan` decorator so PMF at non-integral args is 0".

We might want to floor the argument of other distribution methods so public methods would only call private methods with integer arguments?